### ISSUE TITLE
Linter for length(x == y) where almost certainly length(which(x == y)) or length(x) == y was intended

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1991]
This is a mistake that I find myself making all the time:

```r
if (length(x == 0)) ...
```

It makes itself into a really subtle bug and many wasted minutes.

I can't think of a case where `length(x == y)` is preferable to `length(x)` or `length(y)`, unless there's some weird coercion going on and that should be made explicit anyway.

Therefore it seems like a good candidate for a new linter.

~I don't really know what to name such a linter... I also wonder if maybe it's good for an umbrella linter like `common_mistakes_linter()`, though I can't really think of other good candidate rules for that off the top of my head.~ see comment below

**Comment by AshesITR** [comment]
How about `length_equals_linter()`?
I imagine the most common intention is `length(x) == y`.

I don't think we should make an umbrella linter.

**Comment by MichaelChirico** [comment]
Note to self: the right way to group this is with the tag `common_mistakes`.

**Comment by AshesITR** [comment]
BTW the `length(which())` case should be replaced by `sum()`.

**Comment by MichaelChirico** [comment]
> I don't think we should make an umbrella linter.

Agree, it's better to handle this with a `tag`

**Comment by MichaelChirico** [comment]
> BTW the `length(which())` case should be replaced by `sum()`.

I think that's a separate linter... there's some nuance there, as `length(which(A))` is equivalent to `sum(A)` but the most common case will be comparing `length(which(A))` to `0`, in which case `any(A)` is preferable, which is `boolean_arithmetic_linter()`.
